crates/workspace/src/workspace.rs

```diff
@@ -766,6 +767,7 @@ pub struct Workspace {
     serialized_ssh_project: Option<SerializedSshProject>,
     _items_serializer: Task<Result<()>>,
     session_id: Option<String>,
+    last_edit_location: Option<NavigationEntry>,
```

Comment:
That does not seem needed, given that we have an entire nav history.

Also consider the same feature JetBrains IDEs , where re-invoking the same command over and over moves you back over the edit history — seems to be much more useful to implement it this way instead of a single entry.

Reply:
yeah, that make sense. But after reading editor code, I can't find any obvious signal to distinguish between `edit` change and `select` change, can you give me some cue?

if I use an additional parameter in `Editor::change_selections`, this will need modify many code..., currently not find a better way.

Reply:
Sorry, this is the thing am not aware of (hence did not fix this issue myself).
I just wanted to give a general approach note before this PR went off too far, as I think it's much more beneficial to do it the way close to Intellij* doing.